src/webapi.ts

```diff
@@ -21,6 +21,80 @@ export interface WebApiResponse<Result> {
   severity: string | undefined;
 }
 
+export function updateQuery(query: CmkQuery): void {
+  // convert legacy query (context/params) to new requestSpec interface
+  if (query.context !== undefined && query.params !== undefined) {
+    // we need to replace the object in place, in order to remove the context
+    // and params properties, otherwise both requestSpec and context/params are defined.
+    query.requestSpec = requestSpecFromLegacy(query.context, query.params);
+    delete query.context;
+    delete query.params;
+  }
+}
+
+function transform_negated(
+  context_property: Record<string, string | undefined> | undefined,
+  key_name: string
+): NegatableOption | undefined {
+  if (context_property === undefined) {
+    return undefined;
+  }
+  return {
+    value: context_property[key_name] || '',
+    negated: context_property[`neg_${key_name}`] === 'on',
+  };
+}
+
+export function requestSpecFromLegacy(context: Context, params: Params): RequestSpec {
+  const rs: RequestSpec = {};
+  if (context.host !== undefined) {
+    rs.host_name = context.host.host;
+  }
```

Comment:
This as well as the few ones below could be written as
```
rs.host_name = context.?host.?host
````
couldn't it?
No biggy, it's more consistent with the rest of the file anyway.

Reply:
would keep it as is, i like that all properties are consistent.